src/main/java/com/flexrate/flexrate_back/report/api/ConsumptionHabitReportController.java

```diff
@@ -0,0 +1,56 @@
+package com.flexrate.flexrate_back.report.api;
+
+import com.flexrate.flexrate_back.member.application.MemberService;
+import com.flexrate.flexrate_back.member.domain.Member;
+import com.flexrate.flexrate_back.report.application.ConsumptionHabitReportService;
+import com.flexrate.flexrate_back.report.dto.ConsumptionHabitReportResponse;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+import java.time.YearMonth;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/reports")
+public class ConsumptionHabitReportController {
+
+    private final ConsumptionHabitReportService reportService;
+    private final MemberService memberService;
+
+    /**
+     * 사용자 본인의 소비 습관 리포트 조회 API
+     * @param principal 현재 로그인한 사용자 정보
+     * @param month 조회할 리포트 월 (yyyy-MM 형식, optional)
+     * @return 소비습관 리포트 목록 (단일 or 전체)
+     * @since 2025.05.08
+     * @author 서채연
+     */
+    @Operation(summary = "소비 리포트 조회", description = "특정 월을 입력하면 해당 월 리포트를, 입력하지 않으면 전체 리포트를 반환합니다.")
+    @GetMapping("/consumption-report")
+    public List<ConsumptionHabitReportResponse> getMyReports(
+            Principal principal,
+            @Parameter(description = "조회할 연월 (yyyy-MM)", example = "2025-05")
+            @RequestParam(value = "조회할 연월 (yyyy-MM)", required = false)
+            @DateTimeFormat(pattern = "yyyy-MM") YearMonth month
```

Comment:
변경해야 될 부분들 알려드릴게요!
1. 기존 컨벤션에서는 @Operation에 @Parameter를 붙였으니, @Parameter는 메소드 인자가 아닌 @Operation 위치로 바뀌어야 됩니다!
2. @RequestParam(value = "조회할 연월 (yyyy-MM)", ...) 이렇게 작성하게되면, 실제 쿼리 파라미터명이 "조회할 연월 (yyyy-MM)"이 됩니다. 관례적으로 이 부분에는 영어 snake_case 또는 camelCase를 사용해야 합니다. 현재 코드에서는 변수면(month)와 통일 되도록 "month"로 바꿔주는게 좋아보이네요!

아래에 변경 코드 첨부하겠습니다!

```suggestion
@Operation(
    summary = "소비 리포트 조회",
    description = "특정 월을 입력하면 해당 월 리포트를, 입력하지 않으면 전체 리포트를 반환합니다.",
    parameters = {
        @Parameter(name = "month", description = "조회할 연월 (yyyy-MM)", required = false, example = "2025-05")
    }
)
@GetMapping("/consumption-report")
public List<ConsumptionHabitReportResponse> getMyReports(
        @RequestParam(value = "month", required = false)
        @DateTimeFormat(pattern = "yyyy-MM") YearMonth month,
        Principal principal
```

Reply:
규칙을 어겼었군요 제가ㅎㅎㅎ
꼼꼼하게 봐주셔서 감사합니당 수정했습니다! -> 7852452d0c3695ded445bc7cac1956e1ffd94a26